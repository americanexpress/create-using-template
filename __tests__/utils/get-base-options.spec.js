/*
 * Copyright 2021 American Express Travel Related Services Company, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
 * in compliance with the License. You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under the License
 * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
 * or implied. See the License for the specific language governing permissions and limitations
 * under the License.
 */

const prompts = require('prompts');
const getBaseOptions = require('../../src/utils/get-base-options');

jest.mock('prompts', () => jest.fn(() => 'promptsResponseMock'));

describe('getBaseOptions', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });
  it('should call prompts with the correct set of base options', () => {
    getBaseOptions();
    expect(prompts).toHaveBeenCalledTimes(1);
    // snapshot params as its a large array that will grow over time.
    // test prompts validation
    expect(prompts.mock.calls[0][0][0].validate('testProjectName')).toBe(true);
    expect(prompts.mock.calls[0]).toMatchSnapshot();
  });
  it('should call prompts with the correct set of options when regex is passed', () => {
    const testRegex = /[^a-z-]+/gi;
    getBaseOptions(testRegex);
    expect(prompts).toHaveBeenCalledTimes(1);
    // snapshot params as its a large array that will grow over time.
    // test prompts validation
    expect(prompts.mock.calls[0][0][0].validate('test Project Name')).toBe('Invalid project name format, please make corrections.');
    expect(prompts.mock.calls[0][0][0].format('test Project Name')).toBe("testProjectName");
    expect(prompts.mock.calls[0]).toMatchSnapshot();
  });
});
