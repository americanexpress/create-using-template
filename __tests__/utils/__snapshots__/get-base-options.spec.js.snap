// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getBaseOptions should call prompts with the correct set of base options 1`] = `
Array [
  Array [
    Object {
      "format": [Function],
      "initial": "",
      "message": "Enter your project's name. This will also be used as the directory name for the project:",
      "name": "projectName",
      "type": "text",
      "validate": [Function],
    },
  ],
]
`;

exports[`getBaseOptions should call prompts with the correct set of options when regex is passed 1`] = `
Array [
  Array [
    Object {
      "format": [Function],
      "initial": "",
      "message": "Enter your project's name. This will also be used as the directory name for the project:",
      "name": "projectName",
      "type": "text",
      "validate": [Function],
    },
  ],
]
`;
